packages/flutter_migrate/lib/src/compute.dart

```diff
@@ -0,0 +1,1045 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'base/common.dart';
+import 'base/file_system.dart';
+import 'base/logger.dart';
+import 'base/project.dart';
+import 'custom_merge.dart';
+import 'environment.dart';
+import 'flutter_project_metadata.dart';
+import 'migrate_logger.dart';
+import 'result.dart';
+import 'utils.dart';
+
+// This defines paths of files and directories relative to the project root
+// that should be ignored by the migrate tool regardless of .gitignore and
+// config settings.
+// Paths use `/` as a stand-in for path separator.
+const List<String> _skippedFiles = <String>[
+  'ios/Runner.xcodeproj/project.pbxproj', // Xcode managed configs that may not merge cleanly.
+  'README.md', // changes to this shouldn't be overwritten since is is user owned.
+];
+
+const List<String> _skippedDirectories = <String>[
+  '.dart_tool', // The .dart_tool generated dir.
+  '.git', // Git metadata.
+  'assets', // Common directory for user assets.
+  'build', // Build artifacts.
+  'lib', // Always user owned and we don't want to overwrite their apps.
+  'test', // Typically user owned and flutter-side changes are not relevant.
+];
+
+// Returns true for paths relative to the project root that should be skipped
+// completely by the migrate tool.
+bool _skipped(String localPath, FileSystem fileSystem,
+    {Set<String>? skippedPrefixes}) {
+  for (final String path in _skippedFiles) {
+    if (path.replaceAll('/', fileSystem.path.separator) == localPath) {
```

Comment:
Having just gone down a similar rabbit hole with https://github.com/flutter/flutter/issues/115651, I would suggest using the `canonicalize()` method from package:path rather than doing `replaceAll('/', fileSystem.path.separator)` (this would also make lines 43-45 redundant). Something like:

```dart
  final Iterable<String> canonicalizedSkippedFiles = _skippedFiles.map<String>(
    (String path) => canonicalize(path),
  );
  final String canonicalizedLocalPath = canonicalize(localPath);
  if (canonicalizedSkippedFiles.contains(canonicalizedLocalPath)) {
    return true;
  }
```

Reply:
This will normalize path separators, relative paths, and ASCII case on Windows.

Reply:
Switched to use canonicalise!